packages/netsuite-adapter/src/mapped_lists/mapped_lists.ts

```diff
@@ -0,0 +1,152 @@
+/*
+*                      Copyright 2021 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import {
+  Values, InstanceElement, isMapType, ObjectType, isField, isListType, Field,
+  MapType, isObjectType, createRefToElmWithValue, ListType, isContainerType, ElemID, Value,
+} from '@salto-io/adapter-api'
+import { transformElement, TransformFunc, transformValues } from '@salto-io/adapter-utils'
+import { collections } from '@salto-io/lowerdash'
+import { logger } from '@salto-io/logging'
+import _ from 'lodash'
+import { INDEX, LIST_MAPPED_BY_FIELD, SCRIPT_ID } from '../constants'
+import { listMappedByFieldMapping } from './mapping'
+
+const { awu } = collections.asynciterable
+
+const log = logger(module)
+
+export const convertFieldsTypesToMap = async (
```

Comment:
nit: I think that the naming should be a bit more explicit in this case
```suggestion
export const convertFieldsTypesFromListToMap = async (
```